src/default-config-backend.cpp

```diff
@@ -66,6 +66,13 @@ static int handle_config_updated(int fd, uint32_t mask, void *data)
         // - Config file was created inside parent directory
         should_reload |=
             (event->wd == wd_cfg_file) || (cfg_file_basename == event->name);
+
+        if ((event->name == cfg_file_basename) && (event->wd == wd_cfg_dir))
+        {
+            inotify_rm_watch(fd, wd_cfg_file);
+            wd_cfg_file =
+                inotify_add_watch(fd, (config_dir + "/" + cfg_file_basename).c_str(), IN_CLOSE_WRITE);
```

Comment:
I wonder, do we need to concatenate the two files here, isn't this basically `config_file.c_str()` here?

Reply:
It would be, except in the case where the config file is specified as a relative path. I wanted to make sure the file path we're watching is complete and not rely on relative paths.